code/modules/NTNet/netdata.dm

```diff
@@ -7,6 +7,22 @@
 	var/plaintext_data
 	var/plaintext_data_secondary
 	var/plaintext_passkey
+	var/passkey
+
+// Process data before sending it
+/datum/netdata/proc/pre_send(datum/component/ntnet_interface/interface)
+	// Decode the passkey, if any.
+	if(plaintext_passkey && !passkey)
+		passkey = json_decode(XorEncrypt(hextostr(plaintext_passkey, TRUE), SScircuit.cipherkey))
```

Comment:
maybe there should be another cipherkey for accesses.Because otherwise you can relatively easy reconstruct REF encryption key, become Neo and cheese the Matrix.